### PR TITLE
Also strip padding token for opt

### PR DESCRIPTION
**Pull request by mattdangerw**
Unlike GPT, OPT has a distinct token for padding. It is probably good practice to strip it out doing de-tokenization for generation.

### CODE REVIEW

**Comment by jbischof** [comment]
It seems like you're adding a line that's already there? 🤔 🤔 🤔

**Comment by mattdangerw** [comment]
@jbischof look closer, there's a few chars different. "pad" vs "end"

**Review by jbischof — APPROVED** [review]

**Comment by chenmoneygithub** [comment]
@mattdangerw Thanks! I am a bit curious if OPT model will generate `pad_token`, because technically it should never been used as label in pretraining and finetuning. Or maybe there is some special task needing to generate a `pad_token` to separate the generated segments.. (just my wild guess)?

**Comment by mattdangerw** [comment]
@chenmoneygithub

Looked a bit deeper into this, I think the underlying issue I was seeing is https://github.com/keras-team/keras-nlp/pull/1029

Still an interesting question if we should remove the `<pad>` markers during detokenization. If you were padding your inputs at any point and not using an end of sequence marker, you probably do not want to detokenize all your padding markers. But that becomes a little more niche of a question.

**Comment by chenmoneygithub** [comment]
@mattdangerw Got it, as we have checked in #1029, the truncation problem should be resolved. And yes, it seems to be a separate question if we should truncate out the padding tokens. I vote for yes because we manually convert the prompt inputs into dense for XLA purpose, so it makes sense to me to cut them off.

**Review by chenmoneygithub — APPROVED** [review]
